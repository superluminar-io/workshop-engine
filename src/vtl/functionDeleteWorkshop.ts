export const request = `
  {
    "version" : "2018-05-29",
    "operation" : "DeleteItem",
    "key": {
      "id": $util.dynamodb.toDynamoDBJson($ctx.arguments.workshopId)
    }
  }
`;

export const response = `
  $util.toJson($ctx.result)
`;
