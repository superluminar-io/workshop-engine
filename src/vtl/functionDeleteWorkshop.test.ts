import Parser from 'appsync-template-tester';
import { request, response } from './functionDeleteWorkshop';

describe('request mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(request);

    const context = {
      arguments: {
        workshopId: 'beb79a10-437e-48df-96e2-9cd87fdc063f',
      },
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});

describe('response mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(response);

    const context = {
      result: {
        any: 'data',
      },
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});
