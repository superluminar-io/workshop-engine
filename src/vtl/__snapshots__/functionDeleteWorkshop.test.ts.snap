// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request mapping should match snapshot 1`] = `
Object {
  "key": Object {
    "id": Object {
      "S": "beb79a10-437e-48df-96e2-9cd87fdc063f",
    },
  },
  "operation": "DeleteItem",
  "version": "2018-05-29",
}
`;

exports[`response mapping should match snapshot 1`] = `
Object {
  "any": "data",
}
`;
